VSRAD.Syntax/IntelliSense/Navigation/NavigationTokenServiceResult.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.Collections.Generic;
+using VSRAD.Syntax.Core.Tokens;
+
+namespace VSRAD.Syntax.IntelliSense.Navigation
+{
+    public struct NavigationTokenServiceResult
+    {
+        public static NavigationTokenServiceResult Empty { get { return new NavigationTokenServiceResult(); } }
+
+        public bool HasValue { get; }
```

Comment:
You don't need to explicitly define `HasValue` and `Empty`. Replace `NavigationTokenServiceResult` with `NavigationTokenServiceResult?` where you need nullability or, better yet, convert this to a class (since `Nullable<T>` boxes the value, you don't get the usual benefits of using a struct) and use `null` in place of `Empty`.

Reply:
You are right, the current implementation has a lot of unnecessary operations. Resolved: 5482bea